spec/features/users/logout_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+describe "When a user logs out" do
+  it "they are redirected to the home page, see a flash message indicating they are logged out, and their shopping cart has been cleared" do
+    user = User.create(name: 'Christopher', address: '123 Oak Ave', city: 'Denver', state: 'CO', zip: 80021, email: 'christopher@email.com', password: 'p@ssw0rd', role: 0)
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+    mike = Merchant.create(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
+    paper = mike.items.create(name: "Lined Paper", description: "Great for writing on!", price: 20, image: "https://cdn.vertex42.com/WordTemplates/images/printable-lined-paper-wide-ruled.png", inventory: 3)
+    visit "/items/#{paper.id}"
+    click_on "Add To Cart"
+    visit '/items'
+    within 'nav' do
+      expect(page).to have_content("Cart: 1")
+    end
+    expect(page).to have_content("Logged in as #{user.name}")
+    click_link("Logout")
```

Comment:
The logout process doesn't work when we stub the login. It's good that you figured out that we need to manually log in on the test in order for us to log out. Good work.